Cargo.toml

```diff
@@ -3,6 +3,38 @@
 
 [workspace]
 members = [
+  "alloc",
+  "builder",
+  "crashtracker",
+  "crashtracker-ffi",
+  "profiling",
+  "profiling-ffi",
+  "profiling-replayer",
+  "ddcommon",
+  "ddcommon-ffi",
+  "ddtelemetry",
+  "ddtelemetry-ffi",
+  "tools",
+  "ipc",
+  "ipc/macros",
+  "sidecar",
+  "sidecar/macros",
+  "sidecar-ffi",
+  "tools/cc_utils",
+  "tools/sidecar_mockgen",
+  "trace-normalization",
+  "trace-obfuscation",
+  "trace-utils",
+  "spawn_worker",
+  "tests/spawn_from_lib",
+  "serverless",
+  "bin_tests",
+  "data-pipeline",
+  "data-pipeline-ffi",
+  "ddsketch",
+]
+
+default-members = [
```

Comment:
So this makes `cargo tree` report a number of dependencies as _separate_ instead of the same, ie. `bytes v1.6.0` instead of `bytes v1.6.0 (*)`. I truly don't understand how `cargo` works, and what this setting means for the final artifacts, but have we checked if this affects the sizes?

Reply:
While looking at the benchmark time changes, I also noticed that this `default-members` thing has other side effects like not running `clippy` on `builder` from the root directory unless you add `--workspace`. What is the real reason for having this?

Reply:
The reason for not including the builder in the default-members was to save time and resources in scenarios that don't need to build the artifacts, like binding projects or test scenarios.

Reply:
Removing `--workspace` made no difference to the run-time of the benchmark task. It took 9 minutes 53 seconds, and the last one with `--workspace` took 9 minutes and 50 seconds.